orca-redis/src/main/java/com/netflix/spinnaker/orca/locks/RedisLockManager.java

```diff
@@ -0,0 +1,244 @@
+package com.netflix.spinnaker.orca.locks;
+
+import static net.logstash.logback.argument.StructuredArguments.kv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import redis.clients.jedis.Jedis;
+import redis.clients.util.Pool;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static java.util.Arrays.asList;
+
+@Component
+public class RedisLockManager implements LockManager {
+  private static final int LOCK_VALUE_APPLICATION_IDX = 0;
+  private static final int LOCK_VALUE_TYPE_IDX = 1;
+  private static final int LOCK_VALUE_IDX = 2;
+
+  private static final int JEDIS_POOL_MAX_ATTEMPTS = 3;
+  private static final int JEDIS_POOL_INITIAL_BACKOFF_MILLIS = 10;
+  private static final int JEDIS_POOL_BACKOFF_INCREMENT_MILLIS = 10;
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private final Pool<Jedis> jedisPool;
```

Comment:
Would like to see this use `RedisClientDelegate` instead.

Reply:
+1
Going to look at / probably move the bit of retry logic I have around getting a resource from the pool into kork